src/storage/exec/IndexAggregateNode.cpp

```diff
@@ -0,0 +1,125 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include "storage/exec/IndexAggregateNode.h"
+
+namespace nebula {
+namespace storage {
+
+IndexAggregateNode::IndexAggregateNode(const IndexAggregateNode& node)
+    : IndexNode(node), statInfos_(node.statInfos_), returnColumnsCount_(node.returnColumnsCount_) {
+  stats_ = node.stats_;
+  retColMap_ = node.retColMap_;
+}
+
+IndexAggregateNode::IndexAggregateNode(
+    RuntimeContext* context,
+    const std::vector<std::pair<std::string, cpp2::StatType>>& statInfos,
+    size_t returnColumnsCount)
+    : IndexNode(context, "IndexAggregateNode"),
+      statInfos_(statInfos),
+      returnColumnsCount_(returnColumnsCount) {}
+
+nebula::cpp2::ErrorCode IndexAggregateNode::init(InitContext& ctx) {
```

Comment:
will ctx change? input parameters should usually be values or const references, and input/output parameters should usually be non-const pointers.

Reply:
I guess this because in other node the ctx may be modified, although not modified in this node.

Reply:
this node also modify the context, to give the columns in aggregate but not in returnColumns to IndexScanNode

Reply:
I do not recommend writing this way. If the scalability of the program is considered, expression calculations may be implemented in the projection node later. For example, max(a)-min(a), the subtraction needs to be done in the projection node. If it is not placed in the returnColumn, it will bring additional complexity to the projection node.